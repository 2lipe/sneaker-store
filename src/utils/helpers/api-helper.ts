import axios from 'axios';
import { baseConfig } from '../../config/baseConfig';

export const api = axios.create({
  baseURL: baseConfig.baseApiUrl,
});
