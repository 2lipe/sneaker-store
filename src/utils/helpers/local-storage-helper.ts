/* eslint-disable consistent-return */
import { ISneakers } from '../../models/Interfaces/ISneakers';

export const saveSneakersLocalStorage = (data: ISneakers[]) => {
  if (typeof window !== 'undefined') {
    window.localStorage.setItem('@trustly-store:sneakers', JSON.stringify(data));
  }
};

export const getSneakerLocalStorage = () => {
  if (typeof window !== 'undefined') {
    const sneakersString = window.localStorage.getItem('@trustly-store:sneakers');

    if (sneakersString) {
      const sneakers: ISneakers[] = JSON.parse(sneakersString);

      return sneakers;
    }

    return null;
  }
};
