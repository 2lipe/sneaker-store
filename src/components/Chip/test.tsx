import React from 'react';
import { screen } from '@testing-library/react';
import { ArrowBack } from '@styled-icons/boxicons-regular/ArrowBack';

import { renderWithTheme } from '../../utils/tests/helpers';

import { Chip } from '.';

describe('<Chip />', () => {
  it('should be render the button with text', () => {
    const { container } = renderWithTheme(<Chip text="Back" />);

    expect(screen.getByText(/Back/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should be render an icon version', () => {
    renderWithTheme(<Chip icon={<ArrowBack data-testid="icon" />} text="Back" />);

    expect(screen.getByText(/Back/i)).toBeInTheDocument();
    expect(screen.getByTestId('icon')).toBeInTheDocument();
  });
});
