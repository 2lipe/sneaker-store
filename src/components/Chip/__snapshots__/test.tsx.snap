// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['<Chip /> should be render the button with text 1'] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  outline: none;
  width: 11.5rem;
  height: 3.4rem;
  border-radius: 1.7rem;
  background-color: rgba(0,0,0,0.05);
  -webkit-transition: 0.3s ease-in-out;
  transition: 0.3s ease-in-out;
}

.c0:hover {
  background-color: rgba(0,0,0,0.24000000000000002);
}

.c1 {
  font-family: Arial,Helvetica,sans-serif;
  font-weight: 400;
  font-size: 2.0rem;
  color: #000000;
}

<button
  class="c0"
>
  <p
    class="c1"
  >
    Back
  </p>
</button>
`;
